tuning/acceleration/plugins/framework_plugin_unsloth_autogptq.py

```diff
@@ -0,0 +1,150 @@
+    lora_config: LoraConfig,
+    adapter_name: str,
+    target: torch.nn.Module,
+    **kwargs,
+):
+    # if the base layer module matches a supported class, dispatch the lora linear
+    # to be installed
+    new_module = target
+    if (
+        isinstance(target, GPTQuantLinear)
+    ):
+        new_module = LoraLinearGPTQ(target, adapter_name, **kwargs)
+    # if module cannot be found, return None which results in a raise in the call-stack
+    return new_module
+
+class UnslothAutoGPTQAccelerationPlugin(AccelerationPlugin):
+    
+    require_packages = ['auto_gptq', 'unsloth']
+    restricted_model_archs = ['MixtralForCausalLM', 'LlamaForCausalLM', 'MistralForCausalLM', 'GemmaForCausalLM']
+
+    def __init__(self, configurations: Dict[str, Dict]):
+        super().__init__(configurations)
+
+        # just do checking, nothing must to configure at this point
+        # if need to configure then do something like this:
+        # self.kernel = self._get_config_value("peft.kernel")
+        self._check_config_equal(key="peft.quantization", value="unsloth")
+        self._check_config_equal(key="peft.quantization.unsloth.base_layer", value="auto_gptq")
+        self._check_config_equal(key="peft.quantization.unsloth.kernel", value="triton_v2")
+
+    def model_loader(self, model_name: str, **kwargs):
+        # 1. Load the gptq base model through unsloth FastLanguageModel
+        model, _ = FastLanguageModel.from_pretrained(
+            model_name,
+            quantization_method=QuantizationMethod.GPTQ,
+            device_map="auto",
+        )
+
+        # 2. Depending on the model, replace appropriate CUDA kernel with TritonV2 kernel 
+        if model.config.model_type == 'mixtral':
+            from auto_gptq.nn_modules.qlinear.qlinear_cuda_old import (
+                QuantLinear as QuantLinearCuda,
+            )                
+        else:
+            from auto_gptq.nn_modules.qlinear.qlinear_cuda import (
+                QuantLinear as QuantLinearCuda,
+            )
+        GPTQuantLinear.inject_to_model(
+            model, target_module_type=QuantLinearCuda
+        )
+
+        return model
+
+    @property
+    def requires_custom_loading(self):
+        return True
+
+    @property
+    def requires_agumentation(self):
+        return True
+
+    def augmentation(
+        self, 
+        model, 
+        train_args: TrainingArguments,
+        modifiable_args: Tuple[LoraConfig],
+    ):
+        peft_config, = modifiable_args # unpack modifiable args
+
+        # some assertions
+        assert peft_config is not None, "need peft_config to install PEFT adapters"
+        assert model.dtype == torch.float16 or train_args.fp16,\
+             "need to run in fp16 mixed precision or load model in fp16"
+        peft_kwargs = peft_config.to_dict()
+        assert peft_kwargs['lora_dropout'] == 0, "Unsloth Fused Attention requires lora_dropout argument to be set to 0"
+
+        peft_kwargs.pop('task_type')
```

Comment:
to improve readiblity, i wont convert and pop the 'task type'

1. check peft_config.lora_dropout == 0
2. then when passing kwargs do
```python
func(**{k:v for k,v in peft_config.to_dict() if k != 'task_type'})
```